src/qililab/waveforms/iq_pair.py

```diff
@@ -38,6 +38,9 @@ def __post_init__(self):
         if self.I.get_duration() != self.Q.get_duration():
             raise ValueError("Waveforms of an IQ pair must have the same duration.")
 
+        if not isinstance(self.I, Waveform) or not isinstance(self.Q, Waveform):
```

Comment:
Maybe it's better to make this check first and then check equality of durations? Since we first want to ensure we have the correct class and then call the method of that class.

Reply:
Good suggestion, I made the necessary changes.